fawltydeps/cli_parser.py

```diff
@@ -204,6 +204,16 @@ def populate_parser_paths_options(parser: argparse._ActionsContainer) -> None:
             " (imports), dependency declarations and/or Python environments."
         ),
     )
+    parser.add_argument(
+        "--base-dir",
+        type=Path,
+        metavar="BASE_DIR",
+        help=(
+            "Directory used as base for determining 1st-party imports, i.e."
+            " where your project expects to find its own packages/modules."
+            " Defaults to any directory given to --code (or search_paths)."
```

Comment:
The subtlety is that `--base-dir` applies to all --code/search_paths, while the default is for each --code/search_path to be its own base_dir (when it is a dir).


```suggestion
            "Directory used as base for determining 1st-party imports, i.e."
            " where your project expects to find its own packages/modules."
            " If not provided, each directory given to --code (or search_paths)"
            " acts as the base directory for all its children; files will default"
            " to use the current working directory instead."
```

Reply:
Yeah, I think this subtlety is quite hard to convey in a terse help message on the command line. I'm almost tempted to make leave as it is (short and incomplete), and rather document it more thoroughly in a [FAQ](https://tweag.github.io/FawltyDeps/FAQ/) called "What's the deal with `--base-dir`" or something. 😅

I'll have to think about this a little...

Reply:
I added more documentation in [4b17802](https://github.com/tweag/FawltyDeps/pull/489/commits/4b17802e8af9227f12dca9b36f0241c6272a7ccd), opting to keep the CLI help text a little bit shorter.